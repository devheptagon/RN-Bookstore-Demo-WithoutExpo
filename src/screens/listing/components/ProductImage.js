import React from 'react'
import PropTypes from 'prop-types'
import { Image, StyleSheet } from 'react-native'
import FlexView from '../../components/FlexView'
import Layout from '../../../constants/Layout'
import Theme from '../../../constants/Theme'

const noImage = require('../../../../assets/images/no-image.png')

export class ProductImage extends React.Component {
    state = {        
        width: '100%',
        height: '100%'
    }

    componentDidMount () {
        const { imageLinks, calculateActualSize } = this.props        
        if (calculateActualSize && imageLinks && imageLinks.thumbnail) {
            Image.getSize(
                imageLinks.thumbnail, 
                (width, height) => {
                    const ratio = width / height
                    const newHeight = Math.floor((Layout.window.height - 150) / 2)
                    const newWidth =  Math.floor(newHeight * ratio)                               
                    this.setState({ width: newWidth, height: newHeight})
                }
            )     
        }
    }
    
    render () {
        const { imageLinks } = this.props
        const { width, height } = this.state
        return <FlexView centerAll style={styles.ImageField}>
            <Image
                source={imageLinks && imageLinks.thumbnail ? { uri: imageLinks.thumbnail } : noImage}
                style={[styles.Image,  { width, height }]}
            />
        </FlexView>
    }
}

const styles = StyleSheet.create({
    ImageField: {
        padding: 5
    },
    Image: {         
        borderWidth: Theme.General.Image.BorderWidth, 
        borderColor: Theme.General.Image.BorderColor
    }
})

ProductImage.propTypes = {  
    imageLinks: PropTypes.any, 
    calculateActualSize: PropTypes.bool
}

export default ProductImage 