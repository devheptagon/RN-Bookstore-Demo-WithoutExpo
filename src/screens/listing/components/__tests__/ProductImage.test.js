import React from "react";
import { ProductImage } from "../ProductImage";
import renderer from "react-test-renderer";

test("ProductImage renders correctly", () => {
  const tree = renderer.create(<ProductImage />).toJSON();
  expect(tree).toMatchSnapshot();
});
