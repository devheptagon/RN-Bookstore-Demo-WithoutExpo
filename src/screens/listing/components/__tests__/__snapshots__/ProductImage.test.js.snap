// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "borderColor": "#eee",
          "borderWidth": 2,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
